packages/mobile/src/hooks/useFetchAllFavoritedTrackIds.ts

```diff
@@ -0,0 +1,46 @@
+import type { APIFavorite } from '@audius/common'
+import { decodeHashId, encodeHashId, accountSelectors } from '@audius/common'
+import { useSelector } from 'react-redux'
+import { useAsync } from 'react-use'
+
+import { apiClient } from 'app/services/audius-api-client'
+
+import { useIsOfflineModeEnabled } from './useIsOfflineModeEnabled'
+
+const { getUserId } = accountSelectors
+
+export const useFetchAllFavoritedTrackIds = () => {
+  const currentUserId = useSelector(getUserId)
+  const isOfflineModeEnabled = useIsOfflineModeEnabled()
+  return useAsync(async () => {
+    if (!isOfflineModeEnabled) return []
+    let trackIds: number[] = []
+    let loadMore = true
+    let offset = 0
+    while (loadMore) {
+      const url = apiClient.makeUrl(
```

Comment:
I tried doing this using the common `AudiusAPIClient` but was having trouble. It always used the `/full/` endpoint even when I passed in the `PathType.VersionPath`. Probably missed something or built it wrong; not sure.

Reply:
Strange, I feel like it should work. Did you look at one of the other non-full endpoints as a reference? https://github.com/AudiusProject/audius-client/blob/main/packages/common/src/services/audius-api-client/AudiusAPIClient.ts#L1277

Reply:
Yeah pretty sure I had it just like that. I'm gonna merge this and circle back to it. This works for now and there's some more critical feature work still left on my plate.